src/components/expired-subscription-modal/index.jsx

```diff
@@ -1,25 +1,52 @@
 import {
-  useToggle, AlertModal, Button, ActionRow,
+  useToggle, AlertModal, ActionRow, StatefulButton,
 } from '@openedx/paragon';
 import DOMPurify from 'dompurify';
-import { useSubscriptions } from '../app/data';
+import { useContext, useState } from 'react';
+import { AppContext } from '@edx/frontend-platform/react';
+import { postUnlinkUserFromEnterprise, useEnterpriseCustomer, useSubscriptions } from '../app/data';
 
 const ExpiredSubscriptionModal = () => {
+  const [buttonState, setButtonState] = useState('default');
   const { data: { customerAgreement } } = useSubscriptions();
+  const {
+    authenticatedUser: { email: userEmail },
+  } = useContext(AppContext);
+  const { data: enterpriseCustomer } = useEnterpriseCustomer();
+
   const [isOpen] = useToggle(true);
   if (!customerAgreement?.hasCustomLicenseExpirationMessaging) {
     return null;
   }
+  const onClickHandler = (redirectUrl) => async (e) => {
+    e.preventDefault();
+    setButtonState('pending');
+
+    await postUnlinkUserFromEnterprise(enterpriseCustomer.uuid, userEmail);
+
+    // Redirect immediately
+    window.location.href = redirectUrl;
+    setButtonState('default');
+  };
+  const props = {
+    labels: {
+      default: customerAgreement.buttonLabelInModal,
+      pending: 'Redirecting...',
```

Comment:
nit: fwiw I think we could keep the `pending` label the same as the `default`, e.g. "Continue learning" with the spinner.

```jsx
labels: {
  default: customerAgreement.buttonLabelInModal,
}
```